scenes/cslib_string.txt

```diff
@@ -278,4 +278,39 @@
 	*set char_list_index char_list_len
 	*goto contains_loop
 *set cslib_ret false
-*return
\ No newline at end of file
+*return
+
+
+*comment CONSISTS_OF
+*comment ------------------------------
+*comment Checks if the given string consists
+*comment only of characters in a given list.
+*comment ------------------------------
+*comment 		params:
+*comment 			p_string(string): the string to check for the presence of the given cheracters. 
+*comment 			p_char_list(string): a string of characters allowed in the given string
+*comment 		returns:
+*comment 			(boolean): true if p_string consists only of chars in p_char_list, else false
+*label consists_of
+*params p_string p_char_list
+*set cslib_ret true
+*if (p_char_list = "")
+	*bug cslib_string.consists_of: Character list must not be empty.
+*if (p_string = "")
+	*set cslib_ret false
```

Comment:
shouldn't this be true? 
is "" made of all capitals? all vowels? I would say yes

Reply:
It's nothing: nothing consists of... nothing o.O

Reply:
I agree. An empty string consists of nothing. My original question was when both the given string and the character list consisted of empty strings, should the routine return true? But we have decided an empty character raises an error, so that's that.